kv/range_cache.go

```diff
@@ -228,7 +236,14 @@ func (rdc *rangeDescriptorCache) getCachedRangeDescriptorLocked(key proto.Key) (
 
 	// Check that key actually belongs to range
 	if !rd.ContainsKey(keys.KeyAddress(key)) {
-		return nil, nil
+		if !isReverse {
```

Comment:
No, the edge key inclusion rule is not changed. Because reverse scan is seek backwards from endkey(exclusive) to startkey and if the end key is the range's start key, we need to get the previous range desc.